tests/http_base.py

```diff
@@ -32,8 +41,9 @@ def run_daphne_http(
             # Send it the request. We have to do this the long way to allow
             # duplicate headers.
             conn = HTTPConnection(test_app.host, test_app.port, timeout=timeout)
-            # Make sure path is urlquoted and add any params
-            path = parse.quote(path)
+            if not raw_path:
```

Comment:
I don't like this switch at all - it's ugly and it spread itself through too many parts of the code.

Reply:
I attempted to remove it here but I didn't quite get to a state where the tests were all passing: https://github.com/django/daphne/commit/30dc94b9493cd579a5fe395da3a5d17fbbf6d322